From 984233b59ae6025b6aef029b85a8ced53741f6a3 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Thu, 18 Feb 2016 15:45:21 +0100
Subject: [PATCH] tc: ship filter man pages and refer to them in tc.8

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1286711
Upstream Status: iproute2.git commit a257bc7b4c481
Conflicts:
- Context change due to other missing man pages in RHEL7.
- Dropped all references to flower filter as that does not exist in
  RHEL7.

commit a257bc7b4c481d4b2871508edfccc198d1e56c8a
Author: Phil Sutter <phil@nwl.cc>
Date:   Fri Oct 23 19:47:16 2015 +0200

    tc: ship filter man pages and refer to them in tc.8

    Cc: Thomas Graf <tgraf@suug.ch>
    Cc: Alexey Kuznetsov <kuznet@ms2.inr.ac.ru>
    Cc: Jiri Pirko <jiri@resnulli.us>
    Cc: Patrick McHardy <kaber@trash.net>
    Cc: Werner Almesberger <werner@almesberger.net>
    Signed-off-by: Phil Sutter <phil@nwl.cc>
---
 man/man8/Makefile |  4 +++-
 man/man8/tc.8     | 50 ++++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 53 insertions(+), 1 deletion(-)

diff --git a/man/man8/Makefile b/man/man8/Makefile
index 23ac83e..248ecfd 100644
--- a/man/man8/Makefile
+++ b/man/man8/Makefile
@@ -10,7 +10,9 @@ MAN8PAGES = $(TARGETS) ip.8 arpd.8 lnstat.8 routel.8 rtacct.8 rtmon.8 ss.8 \
 	ip-addrlabel.8 ip-l2tp.8 \
 	ip-maddress.8 ip-monitor.8 ip-mroute.8 ip-neighbour.8 \
 	ip-netns.8 ip-ntable.8 ip-rule.8 ip-tunnel.8 ip-xfrm.8 \
-	ip-tcp_metrics.8 ip-netconf.8 ip-token.8
+	ip-tcp_metrics.8 ip-netconf.8 ip-token.8 \
+	tc-basic.8 tc-cgroup.8 tc-flow.8 tc-fw.8 tc-route.8 \
+	tc-tcindex.8 tc-u32.8
 
 all: $(TARGETS)
 
diff --git a/man/man8/tc.8 b/man/man8/tc.8
index 80eaa3b..fb4efd9 100644
--- a/man/man8/tc.8
+++ b/man/man8/tc.8
@@ -141,6 +141,49 @@ It is important to notice that filters reside
 .B within
 qdiscs - they are not masters of what happens.
 
+The available filters are:
+.TP
+basic
+Filter packets based on an ematch expression. See
+.BR tc-ematch (8)
+for details.
+.TP
+bpf
+Filter packets using (e)BPF, see
+.BR tc-bpf (8)
+for details.
+.TP
+cgroup
+Filter packets based on the control group of their process. See
+. BR tc-cgroup (8)
+for details.
+.TP
+flow
+Flow-based classifier, filtering packets based on their flow (identified by selectable keys). See
+.BR tc-flow (8)
+for details.
+.TP
+fw
+Filter based on fwmark. Directly maps fwmark value to traffic class. See
+.BR tc-fw (8).
+.TP
+route
+Filter packets based on routing table. See
+.BR tc-route (8)
+for details.
+.TP
+rsvp
+Match Resource Reservation Protocol (RSVP) packets.
+.TP
+tcindex
+Filter packets based on traffic control index. See
+.BR tc-index (8).
+.TP
+u32
+Generic filtering on arbitrary packet data, assisted by syntax to abstract common operations. See
+.BR tc-u32 (8)
+for details.
+
 .SH CLASSLESS QDISCS
 The classless qdiscs are:
 .TP
@@ -583,24 +626,31 @@ print rates in IEC units (ie. 1K = 1024).
 .B tc
 was written by Alexey N. Kuznetsov and added in Linux 2.2.
 .SH SEE ALSO
+.BR tc-basic (8),
 .BR tc-bfifo (8),
 .BR tc-cbq (8),
+.BR tc-cgroup (8),
 .BR tc-choke (8),
 .BR tc-codel (8),
 .BR tc-drr (8),
 .BR tc-ematch (8),
+.BR tc-flow (8),
 .BR tc-fq (8),
 .BR tc-fq_codel (8),
+.BR tc-fw (8),
 .BR tc-hfsc (7),
 .BR tc-hfsc (8),
 .BR tc-htb (8),
 .BR tc-pfifo (8),
 .BR tc-pfifo_fast (8),
 .BR tc-red (8),
+.BR tc-route (8),
 .BR tc-sfb (8),
 .BR tc-sfq (8),
 .BR tc-stab (8),
 .BR tc-tbf (8),
+.BR tc-tcindex (8),
+.BR tc-u32 (8),
 .br
 .RB "User documentation at " http://lartc.org/ ", but please direct bugreports and patches to: " <netdev@vger.kernel.org>
 
-- 
1.8.3.1

